lib/fluent/plugin/output.rb

```diff
@@ -597,6 +598,47 @@ def terminate
         super
       end
 
+      def actual_flush_thread_count
+        return 0 unless @buffering
+        return @buffer_config.flush_thread_count unless @as_secondary
+        @primary_instance.buffer_config.flush_thread_count
+      end
+
+      # Run the passed block in the appropriate lock condition for multiple threads and workers.
+      # The lock between workers is made for every `worker_lock_name`.
+      # (For multiple workers, the lock is shared if `worker_lock_name` is the same value).
+      # For multiple threads, `worker_lock_name` is not used, and the lock is shared by all
+      # threads in the same process.
+      def lock_if_need(worker_lock_name)
+        get_worker_lock_if_need(worker_lock_name) do
+          get_flush_thread_lock_if_need do
+            yield
+          end
+        end
+      end
+
+      def get_worker_lock_if_need(name)
```

Comment:
The prefix of the method name seems inconsistent (`get` vs `acquire`).

Reply:
Indeed. I will use `acquire`.

Reply:
Fixed!